stats/awscloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,112 @@
+package awscloudwatch
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type client struct {
+	*cloudwatch.CloudWatch
+	namespace string
+}
+
+// ClientFactory return a function to create client
+func ClientFactory(namespace string) func() (cloudWatchClient, error) {
+	return func() (cloudWatchClient, error) {
+		cw, err := newCloudWatchClient()
+		if err != nil {
+			return nil, err
+		}
+
+		return &client{
+			CloudWatch: cw,
+			namespace:  namespace,
+		}, nil
+	}
+}
+
+// newCloudWatchClient creates a new CloudWatch client
+func newCloudWatchClient() (*cloudwatch.CloudWatch, error) {
+	sess, err := session.NewSession()
```

Comment:
Reading the docs for creating an AWS session (https://godoc.org/github.com/aws/aws-sdk-go/aws/session#hdr-Creating_Sessions), this would load the AWS credentials from `~/.aws/credentials` or use `AWS_*` environment variables? I've only skimmed the SDK docs, but that seems a bit messy to me... :confused: 

Ideally, this output shouldn't rely on the presence of files on the filesystem or read environment variables directly from the OS. For one, that makes it harder to test, but my main concern is that this could get in the way of the future native distributed execution support we want to introduce in k6... Also, the environment variables here aren't prefixed with `K6_`, like the other config options are. But from skimming the AWS SDK docs, I'm not sure we can forbid the SDK code from actually doing all of that magic :confused: 

If the k6 output configuration situation was otherwise perfect (and it's unfortunately not  - https://github.com/loadimpact/k6/issues/587, https://github.com/loadimpact/k6/issues/883 :disappointed: ), I would force the issue, but I'm not sure if we shouldn't accept this as-is and deprecate the direct options in the future? @MStoykov, thoughts?

Reply:
The only way that I see to avoid this issue is duplicate the private [newSession](https://github.com/aws/aws-sdk-go/blob/cb953fcef2462895a21d51884f2e054770b01994/aws/session/session.go#L354). I know this is not an ideal solution, but at the same time I don't see a big problem neither. Looking at what `NewSessionWithOptions` does, for instance, seems that most of the functionality is around more flexible configuration.

Reply:
I am not a big fan of loading things automatically without prefixes ... Not certain if copying the `newSession` will be enough, but if it is ... maybe that is good idea.

On a related note (of copying code) ... I am under the impression that implementing the whole PutMetric call in k6 instead of using the aws-sdk will not be ... all that much code given that it is just http request.